File: app/src/main/java/onsignal/CustomNotificationOpenedHandler.java
package onsignal;

import android.content.Context;
import android.content.Intent;

import androidx.core.app.TaskStackBuilder;

import com.crashlytics.android.Crashlytics;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.onesignal.OSNotificationAction;
import com.onesignal.OSNotificationOpenResult;
import com.onesignal.OneSignal;

import org.json.JSONException;
import org.json.JSONObject;

import activity.ListaGenericaActivity;
import applicaton.PopCornApplication;
import domain.FilmeDB;
import domain.FilmeService;
import elenco.ElencoActivity;
import filme.activity.MovieDetailsActivity;
import info.movito.themoviedbapi.TmdbMovies;
import info.movito.themoviedbapi.model.MovieDb;
import listafilmes.activity.MoviesActivity;
import listaserie.activity.TvShowsActivity;
import main.MainActivity;
import pessoa.activity.FotoPersonActivity;
import pessoa.activity.PersonActivity;
import producao.CrewsActivity;
import produtora.activity.ProdutoraActivity;
import similares.SimilaresActivity;
import site.Site;
import temporada.TemporadaActivity;
import trailer.TrailerActivity;
import tvshow.activity.TvShowActivity;
import utils.Constantes;

import static activity.BaseActivity.getLocale;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.alternative_titles;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.credits;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.images;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.releases;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.similar;
import static info.movito.themoviedbapi.TmdbMovies.MovieMethod.videos;

/**
 * Created by icaro on 16/10/16.
 */

public class CustomNotificationOpenedHandler implements OneSignal.NotificationOpenedHandler {
    private static final String TAG = CustomNotificationOpenedHandler.class.getName();
    private JSONObject jsonData;


    // This fires when a notification is opened by tapping on it.
    @Override
    public void notificationOpened(OSNotificationOpenResult result) {
        Context context = PopCornApplication.getInstance().getBaseContext();
        jsonData = result.notification.payload.additionalData;
        OSNotificationAction.ActionType actionType = result.action.type;


        if (actionType == OSNotificationAction.ActionType.ActionTaken) {
            if ("yes".equals(result.action.actionID)) {
                isButton();
                return;
            }

            if ("no".equals(result.action.actionID)) {
                //
                return;
            }

            if ("talvez".equals(result.action.actionID)) {
                return;
            }
        }

        if (jsonData != null) {

            try {
                JSONObject object = jsonData;
                String action = (String) object.get("action");
                    /*Filme Activity */
                if (action.equals("MovieDetailsActivity") || action.equals("FilmeActivity")) {
                    Intent intent = new Intent(context, MovieDetailsActivity.class);

                    if (object.has("color"))
                        intent.putExtra(Constantes.COLOR_TOP, object.getInt("color"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.FILME_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(MovieDetailsActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }
                /*TvShow Activity */
                if (action.equals("TvshowActivity")) {
                    Intent intent = new Intent(context, TvShowActivity.class);
                    if (object.has("color"))
                        intent.putExtra(Constantes.COLOR_TOP, object.getInt("color"));

                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME_TVSHOW, object.getString("nome"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.TVSHOW_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(TvShowActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }
                /*Person Activity */
                if (action.equals("PersonActivity")) {
                    Intent intent = new Intent(context, PersonActivity.class);

                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME_PERSON, object.getString("nome"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.PERSON_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(PersonActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("FilmesActivity")) {
                    Intent intent = new Intent(context, MoviesActivity.class);
                    // if (object.has("aba")) {    só funciona para NO CINEMA - ARRUMAR
                    // intent.putExtra(Constantes.ABA, object.getInt("id"));
                    TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                    stackBuilder.addParentStack(MainActivity.class);
                    stackBuilder.addNextIntent(intent);
                    stackBuilder.startActivities();
                    // }
                }

                if (action.equals("ListaGenericaActivity")) {
                    Intent intent = new Intent(context, ListaGenericaActivity.class);
                    // Log.d("ListaGenericaActivity", "ListaGenericaActivity");
                    if (object.has("nome"))
                        intent.putExtra(Constantes.LISTA_GENERICA, object.getString("nome"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.LISTA_ID, object.getString("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(ListaGenericaActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("TrailerActivity")) {
                    Intent intent = new Intent(context, TrailerActivity.class);

                    if (object.has("sinopse"))
                        intent.putExtra(Constantes.SINOPSE, object.getString("sinopse"));

                    if (object.has("youtube_key")) {
                        intent.putExtra(Constantes.YOU_TUBE_KEY, object.getString("youtube_key"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(TrailerActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("ElencoActivity")) {
                    Intent intent = new Intent(context, ElencoActivity.class);

                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME, object.getString("nome"));

                    if (object.has("mediatype"))
                        intent.putExtra(Constantes.MEDIATYPE, object.getString("mediatype"));

                    if (object.has("tvseason"))
                        intent.putExtra(Constantes.TVSEASONS, object.getString("tvseason"));

                    if (object.has("id") && object.has("mediatype")) {
                        intent.putExtra(Constantes.ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        intent.putExtra("notification", false);
                        stackBuilder.addParentStack(PersonActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("CrewsActivity")) {
                    Intent intent = new Intent(context, CrewsActivity.class);

                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME, object.getString("nome"));

                    if (object.has("mediatype"))
                        intent.putExtra(Constantes.MEDIATYPE, object.getString("mediatype"));

                    if (object.has("tvseason"))
                        intent.putExtra(Constantes.TVSEASONS, object.getString("tvseason"));

                    if (object.has("id") && object.has("mediatype")) {
                        intent.putExtra(Constantes.ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        intent.putExtra("notification", false);
                        stackBuilder.addParentStack(CrewsActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("SiteActivity")) {
                    Intent intent = new Intent(context, Site.class);

                    if (object.has("url")) {
                        intent.putExtra(Constantes.SITE, object.getString("url"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(Site.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("ProdutoraActivity")) {
                    Intent intent = new Intent(context, ProdutoraActivity.class);

                    if (object.has("id")) {
                        intent.putExtra(Constantes.PRODUTORA_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(ProdutoraActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("SimilaresActivity")) {
                    Intent intent = new Intent(context, SimilaresActivity.class);
                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME_FILME, object.getString("nome"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.FILME_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(SimilaresActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("FotoPersonActivity")) {
                    Intent intent = new Intent(context, FotoPersonActivity.class);

                    if (object.has("nome"))
                        intent.putExtra(Constantes.NOME_PERSON, object.getString("nome"));

                    if (object.has("position"))
                        intent.putExtra(Constantes.POSICAO, object.getInt("position"));

                    if (object.has("id")) {
                        intent.putExtra(Constantes.PERSON_ID, object.getInt("id"));
                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(FotoPersonActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }

                if (action.equals("TvShowsActivity")) {
                    Intent intent = new Intent(context, TvShowsActivity.class);

                    // if (object.has("aba")) {    só funciona para NO CINEM  // intent.putExtra(Constantes.ABA, object.getInt("id"));
                    TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                    stackBuilder.addParentStack(TvShowsActivity.class);
                    stackBuilder.addNextIntent(intent);
                    stackBuilder.startActivities();

                }

                if (action.equals("TemporadaActivity")) {
                    Intent intent = new Intent(context, TemporadaActivity.class);

                    if (object.has("temporada_id") && object.has("tvshow_id")) {
                        intent.putExtra(Constantes.TVSHOW_ID, object.getInt("tvshow_id"));
                        intent.putExtra(Constantes.TEMPORADA_ID, object.getString("temporada_id"));
                        intent.putExtra(Constantes.NOME, object.getString("nome"));
                        intent.putExtra(Constantes.COLOR_TOP, object.getString("color"));

                        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
                        stackBuilder.addParentStack(TemporadaActivity.class);
                        stackBuilder.addNextIntent(intent);
                        stackBuilder.startActivities();
                    }
                }


            } catch (JSONException e) {
                e.printStackTrace();
            }
        } else {
            Intent intent = new Intent(context, MainActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT | Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(intent);
        }

    }

    public void isButton() {
        try {
            String action = (String) jsonData.get("action");

            if (action.equals("FilmeActivity")) {
                final FirebaseDatabase database = FirebaseDatabase.getInstance();
                final FirebaseAuth mAuth = FirebaseAuth.getInstance();

                final int id = jsonData.getInt("id");


                new Runnable() {
                    @Override
                    public void run() {
                        TmdbMovies movies = FilmeService.getTmdbMovies();
                        MovieDb movieDb = movies.getMovie(id, getLocale()
                                        //.toLanguageTag() não funciona na Api 14
                                        + ",en,null"
                                , credits, releases, videos, similar, alternative_titles, images);
                       // Log.d(TAG, "run: "+movieDb.getTitle());

                        FilmeDB filmeDB = new FilmeDB();
                        filmeDB.setIdImdb(movieDb.getImdbID());
                        filmeDB.setId(movieDb.getId());
                        filmeDB.setTitle(movieDb.getTitle());
                        filmeDB.setPoster(movieDb.getPosterPath());

                        DatabaseReference myWatch = database.getReference("users").child(mAuth.getCurrentUser()
                                .getUid()).child("watch")
                                .child("movie");

                        myWatch.child(String.valueOf(id)).setValue(filmeDB);

                      //  Log.d(TAG, "run: "+movieDb.getTitle());
                    }
                }.run();

            }


        }catch (Exception e){
            Crashlytics.logException(e);
        }
    }
}
// The following can be used to open an Activity of your choice.
// Replace - getApplicationContext() - with any Android Context.
// Intent intent = new Intent(getApplicationContext(), YourActivity.class);
// intent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT | Intent.FLAG_ACTIVITY_NEW_TASK);
// startActivity(intent);

// Add the following to your AndroidManifest.xml to prevent the launching of your main Activity
//   if you are calling startActivity above.

/*
        <application ...>
          <meta-data android:name="com.onesignal.NotificationOpened.DEFAULT" android:value="DISABLE" />
        </application>

 */

File: app/src/main/java/onsignal/CustomNotificationOpenedHandler.kt
package onsignal

import lista.ListGenericActivity
import android.content.Context
import android.content.Intent
import androidx.core.app.TaskStackBuilder
import applicaton.PopCornApplication
import com.onesignal.OSNotificationAction.ActionType.*
import com.onesignal.OSNotificationOpenResult
import com.onesignal.OneSignal.*
import filme.activity.MovieDetailsActivity
import lista.movie.activity.MoviesActivity
import lista.tvshow.activity.TvShowsActivity
import main.MainActivity
import org.json.JSONException
import org.json.JSONObject
import pessoa.activity.FotoPersonActivity
import produtora.activity.ProductionActivity
import similares.SimilaresActivity
import site.Site
import temporada.SeasonActivity
import trailer.TrailerActivity
import tvshow.activity.TvShowActivity
import utils.Constant
import utils.Constant.COLOR
import utils.Constant.COLOR_TOP
import utils.Constant.ID
import utils.Constant.LISTA_ID
import utils.Constant.NAME
import utils.Constant.NAV_DRAW_ESCOLIDO
import utils.Constant.NOME_PERSON
import utils.Constant.POSICAO
import utils.Constant.Signal.FOTOPERSON
import utils.Constant.Signal.LISTGENERIC
import utils.Constant.Signal.MOVIE
import utils.Constant.Signal.MOVIESLIST
import utils.Constant.Signal.PRODUTORA
import utils.Constant.Signal.SIMILARES
import utils.Constant.Signal.SITE
import utils.Constant.Signal.TEMPORADA
import utils.Constant.Signal.TRAILER
import utils.Constant.Signal.TVSHOW
import utils.Constant.Signal.TVSHOWLIST
import utils.Constant.Signal.VIDEO
import utils.Constant.TEMPORADA_ID
import utils.Constant.TVSHOW_ID

/**
 * Created by icaro on 16/10/16.
 */
class CustomNotificationOpenedHandler : NotificationOpenedHandler {
	private lateinit var context: Context
	private var jsonData: JSONObject? = null

	// This fires when a notification is opened by tapping on it.
	override fun notificationOpened(result: OSNotificationOpenResult) {
		context = PopCornApplication.getInstance().baseContext
		jsonData = result.notification.payload.additionalData
		val actionType = result.action.type
		if (actionType == ActionTaken) {
			if ("yes" == result.action.actionID) {
				return
			}
			if ("no" == result.action.actionID) {
				//
				return
			}
			if ("talvez" == result.action.actionID) {
				return
			}
		}
		if (jsonData != null) {
			try {
				val jsonObject: JSONObject = jsonData as JSONObject

				when (jsonObject.getString("action")) {
                    MOVIE -> {
                        callMovieDetails(context, jsonObject)
                    }
                    TVSHOW -> {
                        callTvShow(context, jsonObject)
                    }
                    MOVIESLIST -> {
                        callMoviesList(context, jsonObject)
                    }
                    LISTGENERIC -> {
                        callGenericList(context, jsonObject)
                    }
                    TRAILER, VIDEO -> {
                        callTrailer(context, jsonObject)
                    }
                    SITE -> {
                        callSite(context, jsonObject)
                    }
                    PRODUTORA -> {
                        callProdutora(context, jsonObject)
                    }
                    SIMILARES -> {
                        callSimilares(context, jsonObject)
                    }
                    FOTOPERSON -> {
                        callPersonPhoto(context, jsonObject)
                    }
                    TVSHOWLIST -> {
                        callTvshowList(context, jsonObject)
                    }
                    TEMPORADA -> {
                        callSeason(context, jsonObject)
                    }
				}
			} catch (e: JSONException) {
				e.printStackTrace()
				callMain()
			}
		} else {
			callMain()
		}
	}

	private fun callMain() {
		val intent = Intent(context, MainActivity::class.java)
		intent.flags = Intent.FLAG_ACTIVITY_REORDER_TO_FRONT or Intent.FLAG_ACTIVITY_NEW_TASK
		context.startActivity(intent)
	}

	private fun callSeason(context: Context, jsonObject: JSONObject) { //todo feito
		if (jsonObject.has(TEMPORADA_ID) && jsonObject.has(TVSHOW_ID)) {
			val intent = Intent(context, SeasonActivity::class.java).apply {
				putExtra(TVSHOW_ID, jsonObject.getInt(TVSHOW_ID))
				putExtra(TEMPORADA_ID, jsonObject.getInt(TEMPORADA_ID))
				putExtra(NAME, jsonObject.getString(NAME))
				putExtra(COLOR_TOP, jsonObject.getString(COLOR_TOP))
			}
			TaskStackBuilder.create(context).apply {
				addParentStack(SeasonActivity::class.java)
				addNextIntent(intent)
				startActivities()
			}
		} else {
			callMain()
		}
	}

	private fun callTvshowList(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, TvShowsActivity::class.java)
		if (jsonObject.has(NAV_DRAW_ESCOLIDO)) {
			// só funciona para NO CINEM
			intent.putExtra(NAV_DRAW_ESCOLIDO, jsonObject.getString(NAV_DRAW_ESCOLIDO));
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(TvShowsActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callPersonPhoto(context: Context, jsonObject: JSONObject) { // Quebrado
		val intent = Intent(context, FotoPersonActivity::class.java)
		if (jsonObject.has(ID)) {
			intent.putExtra(Constant.PERSON_ID, jsonObject.getInt(ID))
			if (jsonObject.has(NOME_PERSON)) intent.putExtra(NOME_PERSON,
                jsonObject.getString(NOME_PERSON))
			if (jsonObject.has(POSICAO)) intent.putExtra(POSICAO, jsonObject.getInt(POSICAO))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(FotoPersonActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callSimilares(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, SimilaresActivity::class.java)
		if (jsonObject.has("nome")) intent.putExtra(Constant.NOME_FILME,
            jsonObject.getString("nome"))
		if (jsonObject.has("id")) {
			intent.putExtra(Constant.FILME_ID, jsonObject.getInt("id"))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(SimilaresActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callProdutora(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, ProductionActivity::class.java)
		if (jsonObject.has("id")) {
			intent.putExtra(Constant.ID, jsonObject.getInt("id"))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(ProductionActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callSite(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, Site::class.java)
		if (jsonObject.has("url")) {
			intent.putExtra(Constant.SITE, jsonObject.getString("url"))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(Site::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callTrailer(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, TrailerActivity::class.java)
		if (jsonObject.has("sinopse")) intent.putExtra(Constant.SINOPSE,
            jsonObject.getString("sinopse"))
		if (jsonObject.has("youtube_key")) {
			intent.putExtra(Constant.YOU_TUBE_KEY, jsonObject.getString("youtube_key"))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(TrailerActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callGenericList(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, ListGenericActivity::class.java)
		if (jsonObject.has(LISTA_ID)) {
			intent.putExtra(LISTA_ID, jsonObject.getString(LISTA_ID))
			if (jsonObject.has(NAME)) intent.putExtra(Constant.LISTA_NOME,
                jsonObject.getString(NAME))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(ListGenericActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callMoviesList(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, MoviesActivity::class.java)
		if (jsonObject.has(NAV_DRAW_ESCOLIDO)) {
			intent.putExtra(NAV_DRAW_ESCOLIDO, jsonObject.getString(NAV_DRAW_ESCOLIDO))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(MainActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callTvShow(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, TvShowActivity::class.java)
		if (jsonObject.has(ID)) {
			intent.putExtra(TVSHOW_ID, jsonObject.getInt(ID))
			if (jsonObject.has(COLOR)) intent.putExtra(COLOR, jsonObject.getInt(COLOR))
			if (jsonObject.has(NAME)) intent.putExtra(Constant.NOME_TVSHOW,
                jsonObject.getString(NAME))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(TvShowActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}

	private fun callMovieDetails(context: Context, jsonObject: JSONObject) {
		val intent = Intent(context, MovieDetailsActivity::class.java)
		if (jsonObject.has(ID)) {
			intent.putExtra(Constant.FILME_ID, jsonObject.getInt(ID))
			if (jsonObject.has(COLOR)) intent.putExtra(COLOR, jsonObject.getInt(COLOR))
			val stackBuilder = TaskStackBuilder.create(context)
			stackBuilder.addParentStack(MovieDetailsActivity::class.java)
			stackBuilder.addNextIntent(intent)
			stackBuilder.startActivities()
		} else {
			callMain()
		}
	}
}